### ISSUE TITLE
Annotation_map does not work with coord_sf

### ISSUE THREAD

**Issue by markpayneatwork** [tidyverse/ggplot2#3653]
When using`coord_sf()` to change the projection of a map, the background layer produced by  `annotation_map()` is not getting reprojected.

First, a plot  without changing projection, using the annotation_map layer to give more context to North Carolina
```r
library(ggplot2)
library(sf)
nc <- sf::st_read(system.file("shape/nc.shp", package = "sf"), quiet = TRUE)

ggplot() +
  annotation_map(map_data("state"),fill="antiquewhite",colour="darkgrey")+
  geom_sf(data = nc)
```
![image](https://user-images.githubusercontent.com/4990832/70135829-1a8dfe80-168b-11ea-96d1-6734c4480b6b.png)

However, changing the projection looses the annotation layer, but reprojects the geometry layer
```r
ggplot() +
  annotation_map(map_data("state"),fill="antiquewhite",colour="darkgrey")+
  geom_sf(data = nc)+
  coord_sf(crs = st_crs(3347))
```
![image](https://user-images.githubusercontent.com/4990832/70135882-32658280-168b-11ea-8042-80ee3df9e653.png)

It looks to me like the annotation_layer is not being reprojected.

**Comment by clauswilke** [comment]
This happens because `annotation_map()` is old code that predates `geom_sf()`. Only sf layers are reprojected. While it would be nice to reproject other layers, there are numerous technical hurdles that make me think it's not going to happen anytime soon.

At this point, the best way forward is probably to amend the documentation accordingly.

**Comment by thomasp85** [comment]
@clauswilke has this been fixed by #3659?

**Comment by clauswilke** [comment]
Yes, correct, just need to set the appropriate default crs. This particular case is actually included in the examples, so I think we're good as is.

``` r
library(ggplot2)
library(sf)
#> Linking to GEOS 3.8.1, GDAL 3.2.1, PROJ 7.2.1
nc <- sf::st_read(system.file("shape/nc.shp", package = "sf"), quiet = TRUE)

ggplot() +
  annotation_map(map_data("state"),fill="antiquewhite",colour="darkgrey")+
  geom_sf(data = nc) +
  coord_sf(crs = st_crs(3347), default_crs = st_crs(4326))
```

![](https://i.imgur.com/Z2gQ2Rh.png)

<sup>Created on 2022-05-18 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>
